torch/nn/parallel/scatter_gather.py

```diff
@@ -24,10 +24,9 @@ def scatter_map(obj):
     # to a closure that has a reference to the scatter_map cell (because the
     # fn is recursive). To avoid this reference cycle, we set the function to
     # None, clearing the cell
-    try:
-        return scatter_map(inputs)
-    finally:
-        scatter_map = None
+    res = scatter_map(inputs)
```

Comment:
What if `scatter_map` throws an exception, will this still be able to clear the reference cycle?